src/Symfony/Component/Security/Http/Firewall/UsernamePasswordJsonAuthenticationListener.php

```diff
@@ -75,10 +75,15 @@ public function __construct(TokenStorageInterface $tokenStorage, AuthenticationM
     public function handle(GetResponseEvent $event)
     {
         $request = $event->getRequest();
-        if (false === strpos($request->getRequestFormat(), 'json')
-            && false === strpos($request->getContentType(), 'json')
-        ) {
-            return;
+        
+        try {
+            if (false === strpos($request->getRequestFormat(), 'json')
+                && false === strpos($request->getContentType(), 'json')
+            ) {
+                throw new BadRequestHttpException('Invalid content type');
+            }
+        } catch (BadRequestHttpException $e) {
+                $request->setRequestFormat('json');
```

Comment:
You can configure more than one listener per firewall, returning early is needed to give a chance to other listeners. Forcing the request format is not good here, and throwing is not an option since this listener is just one piece of the chain.
The client doesn't know what happens because actually, the request does not match any of the configured authentication listeners, that's by design.

Reply:
Thank you for your answer, indeed you are so right :) . Indeed the error response is not really meaningful that's why I was trying to add some hints for the client but wrongly I can see it.